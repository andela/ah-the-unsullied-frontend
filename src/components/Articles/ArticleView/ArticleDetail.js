import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Col, Preloader, Row } from 'react-materialize';
import {getArticle}from '../../../actions/articles/articleActions';
import ArticleView from './views/ArticleView';
import ArticleDelete from './ArticleDelete';
import Nav from '../../common/nav';
import SocialShare from '../../../components/SocialSharing/shareArticle';
import Rating from '../../../components/Rating/RatingComponent';
import LikeDislike from '../LikeDislikeArticle';
import BookmarkArticle from './BookmarkArticle';

export class ArticleDetail extends Component {
  state = {
    fetched: false
  };

  componentDidMount = () => {
    const slug = this.props.match.params.slug;
    this.props.getArticle(slug).then(() => {
      this.setState({
        fetched: true
      });
    });
  };

  render() {
    if (!this.state.fetched) {
      return (
        <div className="container">
          <Row>
            <Col s={5}>
              <Preloader size="big" />
            </Col>
          </Row>
        </div>
      );
    } else {
      const slug = this.props.article.article.article.slug;
      const rating = this.props.article.article.article.rating;
      const shareTitle = this.props.article.article.article.title;
      const shareSlug = this.props.match.url;
      return (
        <div>
          <Nav />
          <BookmarkArticle {...this.props} />
          <ArticleView {...this.props} />
          <Rating slug={slug} rating={rating} />
          <ArticleDelete {...this.props} />
          <LikeDislike {...this.props} />
          <SocialShare title={shareTitle} slug={shareSlug} />
        </div>
      );
    }
  }
}



ArticleDetail.propTypes = {
  getArticles: PropTypes.func.isRequired,
  getArticle: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  article: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
    article: state.articlereducer,
    auth: state.auth
})

export default connect(
  mapStateToProps,
 {getArticle}
)(ArticleDetail);
