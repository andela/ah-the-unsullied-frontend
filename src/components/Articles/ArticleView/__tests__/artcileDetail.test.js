import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ArticleDetail, mapStateToProps } from '../ArticleDetail';

Enzyme.configure({ adapter: new Adapter() });
describe('single article', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
        user: {
          username: 'nesh'
        }
      },

      article: {
        title: 'my article',
        description: 'some stuff',
        slug: 'my-article',
        body: 'gsgsgsg',
        dislikes: 1,
        likes: 2,
        liked: false,
        displiked: false,
        is_reported: false,
        created_at: '01-01-2019',
        author: {
          bio: 'i have abio',
          username: 'nesh',
          image: 'hshhshs'
        }
      },

      match: {
        params: {
          slug: 'my-article'
        }
      },

        getArticle: async () =>{
            await jest.fn()
        }
    };

    state = {
      auth: {},
      articlereducer: {
        article: {
          article: {
            article:{}
          }
        }
      }, 
      loading: true,
    };

    wrapper = shallow(<ArticleDetail {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should mapStateToProps', () => {
    const props = mapStateToProps(state);
    expect(props.article.article).toEqual(state.articlereducer.article);
  });

  

  it('should render article view component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
