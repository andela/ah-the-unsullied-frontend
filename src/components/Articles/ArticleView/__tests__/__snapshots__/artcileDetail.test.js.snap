// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single article should render article view component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleDetail
    article={
      Object {
        "author": Object {
          "bio": "i have abio",
          "image": "hshhshs",
          "username": "nesh",
        },
        "body": "gsgsgsg",
        "created_at": "01-01-2019",
        "description": "some stuff",
        "dislikes": 1,
        "displiked": false,
        "is_reported": false,
        "liked": false,
        "likes": 2,
        "slug": "my-article",
        "title": "my article",
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    getArticle={[Function]}
    match={
      Object {
        "params": Object {
          "slug": "my-article",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row>
        <Col
          s={5}
        >
          <Preloader
            active={true}
            color="blue"
            flashing={false}
            size="big"
          />
        </Col>
      </Row>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          s={5}
        >
          <Preloader
            active={true}
            color="blue"
            flashing={false}
            size="big"
          />
        </Col>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Preloader
            active={true}
            color="blue"
            flashing={false}
            size="big"
          />,
          "s": 5,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "color": "blue",
            "flashing": false,
            "size": "big",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row>
          <Col
            s={5}
          >
            <Preloader
              active={true}
              color="blue"
              flashing={false}
              size="big"
            />
          </Col>
        </Row>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            s={5}
          >
            <Preloader
              active={true}
              color="blue"
              flashing={false}
              size="big"
            />
          </Col>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Preloader
              active={true}
              color="blue"
              flashing={false}
              size="big"
            />,
            "s": 5,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "color": "blue",
              "flashing": false,
              "size": "big",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
